drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -233,8 +236,18 @@ def studio_view(self, context):
         for js_url in js_urls:
             fragment.add_javascript_url(self.runtime.local_resource_url(self, js_url))
 
+        # Do a bit of manipulation so we get the appearance of a list of zone options on
+        # items that still have just a single zone stored
+        data = copy.deepcopy(self.data)
```

Comment:
So, we definitely can, and it'd work. However, as it stands, we already get the new settings as soon as the instructor hits "save" in the settings dialog - it's a POST, not a PATCH, so `data` gets overridden entirely, including removing items that aren't in the POST (i.e., `item['zone']`). Given that, I'd prefer to only actually make the change in the database once the user's explicitly made a decision to save.

Obviously, that's something that can go either way, especially since the user isn't consciously making a choice to move to the new format when they hit Save, but I'd prefer to avoid making the change completely silently just because the Studio view was opened. Seems to go against the "GET doesn't change data" rule.